src/show/components/TextField/index.js

```diff
@@ -1,23 +1,78 @@
-import React from "react";
-
+import React, { useState } from 'react';
+import { EyeIcon } from "../../assets/svg";
 const TextField = ({
+  name = "name",
+  type = 'text',
   label = "",
   placeholder = "",
-  required = false,
   value = "",
-  password = false,
-}) => (
-  <div>
-    <h6>{label}</h6>
-    <input
-      placeholder={placeholder}
-      type={password ? "password" : "text"}
-      password={true}
-      className="bg-gray-100 text-black w-64 px-2 rounded border-2 border-black"
-    //   value={value}
-    />
-    <h6 className="text-xs text-right">{required && "required"}</h6>
-  </div>
-);
-
-export default TextField;
+  onChange = () => {},
+}) => {
+  const [error, setError] = useState(''); 
+  const [inputType, setInputType] = useState(type);
+
+  const validate = (type, value) => {
```

Comment:
move this function outside of the TextField

Reply:
when i moved this function outside of the  textfield,  i got an error   "useState" cannot be called at the top level. then how to move outside.